### DISCUSSION TITLE
About the way equilibrium_detection is computed

### DISCUSSION THREAD

**Discussion by xiki-tempula** [alchemistry/alchemlyb#220] [General]
In the pymbar doc (https://pymbar.readthedocs.io/en/master/timeseries.html#automatically-identifying-the-equilibrated-production-region)

The recommended way of doing the equilibrium_detection is to use pymbar.detectEquilibration first and then pymbar.subsampleCorrelatedData.

In https://alchemlyb.readthedocs.io/en/latest/preprocessing/alchemlyb.preprocessing.subsampling.html#alchemlyb.preprocessing.subsampling.statistical_inefficiency, we did the calculation in the same way of doing pymbar.statisticalInefficiency followed by pymbar.subsampleCorrelatedData.

So why in equilibrium_detection, we did pymbar.detectEquilibration and then do

```
        # we round up
        statinef = int(np.rint(statinef))

        # subsample according to statistical inefficiency
        series = series.iloc[t::statinef]

        df = df.loc[series.index]
```
instead of pymbar.subsampleCorrelatedData? I know they are the same (or similar) but why won't we just use pymbar.subsampleCorrelatedData? It will make the code more clear @dotsdl .

**Comment by orbeckst** [comment]
Just looking at the code, I find slicing of dataframes clearer than calling subsampleCorrelatedData. Perhaps the other reason is that pymbar deals with np arrays and not pandas.DataFrame? (Also, when switching to pymbar 4 we won't have to change that function call...). 

But I'll let @dotsdl chime in.
